File: http/http-server-annotation-processor/src/test/java/ru/tinkoff/kora/http/server/annotation/processor/controller/TestControllerHeaderParameters.java
package ru.tinkoff.kora.http.server.annotation.processor.controller;

import ru.tinkoff.kora.http.common.annotation.Header;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.http.server.common.annotation.HttpController;

import javax.annotation.Nullable;
import java.util.List;
import java.util.Optional;

import static ru.tinkoff.kora.http.common.HttpMethod.GET;

@HttpController
public class TestControllerHeaderParameters {
    /*
    Headers: String, Integer, List<String>, List<Integer>
     */

    @HttpRoute(method = GET, path = "/headerString")
    public String headerString(@Header(value = "string-header") String string) {
        return string;
    }

    @HttpRoute(method = GET, path = "/headerNullableString")
    void headerNullableString(@Header @Nullable String string) {
    }

    @HttpRoute(method = GET, path = "/headerOptionalString")
    void headerNullableString(@Header Optional<String> string) {
    }

    @HttpRoute(method = GET, path = "/headerStringList")
    void headerNullableString(@Header List<String> string) {
    }

    @HttpRoute(method = GET, path = "/headerInteger")
    public void headerInteger(@Header(value = "integer-header") Integer integer) {
    }

    @HttpRoute(method = GET, path = "/headerNullableInteger")
    public void headerNullableInteger(@Header(value = "integer-header") @Nullable Integer integer) {
    }

    @HttpRoute(method = GET, path = "/headerOptionalInteger")
    public void headerOptionalInteger(@Header(value = "integer-header") Optional<Integer> integer) {
    }

    @HttpRoute(method = GET, path = "/headerIntegerList")
    public void headerStringList(@Header(value = "integer-header") List<Integer> integers) {
    }
}

File: http/http-server-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/server/symbol/processor/controllers/TestControllerHeaderParameters.kt
package ru.tinkoff.kora.http.server.symbol.processor.controllers

import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.Header
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.http.server.common.annotation.HttpController

@HttpController
open class TestControllerHeaderParameters {
    /*
    Headers: String / List<String>
     */
    @HttpRoute(method = HttpMethod.GET, path = "/headerString")
    open fun headerString(@Header(value = "string-header") string: String): String {
        return string
    }

    @HttpRoute(method = HttpMethod.GET, path = "/headerNullableString")
    open fun headerNullableString(@Header string: String?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/headerStringList")
    open fun headerNullableString(@Header string: List<String>?) {
    }

    @HttpRoute(method = HttpMethod.GET, path = "/headerInt")
    open fun headerInt(@Header(value = "int-header") intHeader: Int): Int {
        val intx2 = intHeader*2
        return intHeader
    }
}
